src/NServiceBus.Core/Recoverability/FirstLevelRetries/ProcessingFailureInfo.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus
+{
+    using System;
+
+    class ProcessingFailureInfo
```

Comment:
@MarcinHoppe In my opinion if we want to share the types then:
 * they should be moved to a different directory
 * we should do that after other recoverability parts are in place to have enough data to make that decission